docs/components/index.md

```diff
@@ -0,0 +1,360 @@
+
+```cshtml
+<button class="btn btn-primary" onclick="@UpdateHeading">
+    Update heading
+</button>
+
+@functions {
+    void UpdateHeading(UIMouseEventArgs e)
+    {
+        ...
+    }
+}
+```
+
+The following code calls the `CheckboxChanged` method when the check box is changed in the UI:
+
+```cshtml
+<input type="checkbox" class="form-check-input" 
+    onchange="@CheckboxChanged" />
+
+@functions {
+    void CheckboxChanged()
+    {
+        ...
+    }
+}
+```
+
+For some events, event-specific event argument types are permitted. If access to one of these event types isn't necessary, it isn't required in the method call.
+
+The list of supported event arguments is:
+
+* UIEventArgs
+* UIChangeEventArgs
+* UIKeyboardEventArgs
+* UIMouseEventArgs
+
+Lambda expressions can also be used:
+
+```cshtml
+<button onclick="@(e => Console.WriteLine("Hello, world!"))">Say hello</button>
+```
+
+## Lifecycle methods
+
+`OnInitAsync` and `OnInit` execute code after the component has been initialized. To perform an asynchronous operation, use `OnInitAsync` and use the `await` keyword:
+
+```csharp
+protected override async Task OnInitAsync()
+{
+    await ...
+}
+```
+
+For a synchronous operation, use `OnInit`:
+
+```csharp
+protected override void OnInit()
+{
+    ...
+}
+```
+
+`OnParametersSetAsync` and `OnParametersSet` are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after `OnInit` during component initialization.
+
+```csharp
+protected override async Task OnParametersSetAsync()
+{
+    await ...
+}
+```
+
+```csharp
+protected override void OnParametersSet()
+{
+    ...
+}
+```
+
+`SetParameters` can be overridden to execute code before parameters are set:
+
+```csharp
+public override void SetParameters(ParameterCollection parameters)
+{
+    ...
+
+    base.SetParameters(parameters);
+}
+```
+
+If `base.SetParameters` isn't invoked, the custom code can interpret the incoming parameters value in any way required. For example, the incoming parameters aren't required to be assigned to the properties on the class.
+
+`ShouldRender` can be overridden to suppress refreshing of the UI. If the implementation returns `true`, the UI is refreshed. Even if `ShouldRender` is overridden, the component is always initially rendered.
+
+```csharp
+protected override bool ShouldRender()
+{
+    var renderUI = true;
+
+    return renderUI;
+}
+```
+
+## Component disposal with IDisposable
+
+If a component implements [IDisposable](/dotnet/api/system.idisposable), the [Dispose method](https://docs.microsoft.com/dotnet/standard/garbage-collection/implementing-dispose) is called to release unmanaged resources when the component is removed from the UI. A component can implement `IDisposable` in a C# base class that implements `BlazorComponent`. The following component uses `@implements IDisposable` and the `Dispose` method:
```

Comment:
Not a fan of the phrase "release unmanaged resources". I know that's used elsewhere in .NET docs but it's wrong. Disposal is a more general programming pattern than that; you can release arbitrary things or perform other teardown logic inside the dispose call. Can we just say "dispose is called when the component is removed from the UI"?

Reply:
> A component can implement `IDisposable` in a C# base class that implements `BlazorComponent`.

That's a bit of a non-sequitor, as it doesn't relate to the subsequent code sample. Can we remove that sentence?